main.go

```diff
@@ -116,37 +152,87 @@ func main() {
 	m := &Miner{
 		config:         config,
 		engine:         blake3Engine,
-		sliceClients:   connectToSlice(config),
 		header:         types.EmptyHeader(),
-		updateCh:      make(chan *types.Header, resultQueueSize),
+		updateCh:       make(chan *types.Header, resultQueueSize),
 		resultCh:       make(chan *types.Header, resultQueueSize),
-		previousNumber: [common.HierarchyDepth]uint64{0,0,0},
+		previousNumber: [common.HierarchyDepth]uint64{0, 0, 0},
 	}
 	log.Println("Starting Quai cpu miner in location ", config.Location)
-	m.fetchPendingHeader()
-	go m.subscribePendingHeader()
-	go m.resultLoop()
+	if config.Proxy {
+		m.proxyClient = connectToProxy(config)
+		go m.fetchPendingHeader()
```

Comment:
IIUC `fetchPendingHeader()` and `fetchNodeHeader()` are effectlively the same method, just with a different backend (node vs proxy). Maybe rename the first to `fetchProxyHeader()`? Or combine the two with an arg to select which backend?

Reply:
Same comment for `subscribeProxy()` and `subscribePendingHeader()`

Reply:
Done